src/cases/resolvers/activity.resolver.ts

```diff
@@ -1,18 +1,28 @@
 import { Injectable } from '@angular/core';
-
+import { Resolve } from '@angular/router';
 import { ActivityService } from '@hmcts/ccd-case-ui-toolkit';
+import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
 import { Observable, of } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { AppConstants } from '../../app/app.constants';
 
 @Injectable()
-export class ActivityResolver {
-  constructor(private readonly activityService: ActivityService) {}
+export class ActivityResolver implements Resolve<boolean> {
+  constructor(
+    private readonly activityService: ActivityService,
+    private readonly featureToggleService: FeatureToggleService
+  ) {
+    this.featureToggleService.getValue<string>(AppConstants.ACTIVITY_TRACKER_MODE, undefined)
+      .pipe(filter((mode) => !!mode))
+      .subscribe((mode) => {
+        this.activityService.mode = ActivityService.MODES[mode] || ActivityService.MODES.off;
+        if (this.activityService.mode !== ActivityService.MODES.off) {
+          this.activityService.verifyUserIsAuthorized();
+        }
+      });
+  }
 
   public resolve(): Observable<boolean> {
-    try {
-      this.activityService.verifyUserIsAuthorized();
-    } catch (err) {
-      console.log(err);
-    }
-    return of(true);
+    return of(this.activityService.mode !== ActivityService.MODES.off);
```

Comment:
This boolean is referred to twice - should be a boolean variable

Reply:
Hasn't actually been replaced here though right? Should be return of(this.isActivityEnabled())?